server/users.go

```diff
@@ -1,317 +1,153 @@
 package server
 
 import (
-	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"net/http"
-	"net/url"
 
 	"github.com/bouk/httprouter"
 	"github.com/influxdata/chronograf"
 )
 
-// NewSourceUser adds user to source
-func (h *Service) NewSourceUser(w http.ResponseWriter, r *http.Request) {
-	var req userRequest
-	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
-		invalidJSON(w, h.Logger)
-		return
-	}
-
-	if err := req.ValidCreate(); err != nil {
-		invalidData(w, err, h.Logger)
-		return
-	}
-
-	ctx := r.Context()
-	srcID, ts, err := h.sourcesSeries(ctx, w, r)
-	if err != nil {
-		return
-	}
-
-	store := ts.Users(ctx)
-	user := &chronograf.User{
-		Name:        req.Username,
-		Passwd:      req.Password,
-		Permissions: req.Permissions,
-		Roles:       req.Roles,
-	}
-
-	res, err := store.Add(ctx, user)
-	if err != nil {
-		Error(w, http.StatusBadRequest, err.Error(), h.Logger)
-		return
-	}
-
-	if err != nil {
-		Error(w, http.StatusBadRequest, err.Error(), h.Logger)
-		return
-	}
-
-	su := newUserResponse(srcID, res.Name).WithPermissions(res.Permissions)
-	if _, hasRoles := h.hasRoles(ctx, ts); hasRoles {
-		su.WithRoles(srcID, res.Roles)
-	}
-	w.Header().Add("Location", su.Links.Self)
-	encodeJSON(w, http.StatusCreated, su, h.Logger)
+type userRequest struct {
+	ID       uint64 `json:"id"`
+	Name     string `json:"name"`
+	Provider string `json:"provider,omitempty"`
```

Comment:
I'm going back and forth on this... should provider and scheme really be omitempty?

Reply:
Ok